apm-agent-plugins/apm-spring-webflux/apm-spring-webflux-plugin/src/main/java/co/elastic/apm/agent/springwebflux/TransactionAwareSubscriber.java

```diff
@@ -99,17 +60,35 @@
      * @param description human-readable description to make debugging easier
      */
     TransactionAwareSubscriber(CoreSubscriber<? super T> subscriber,
-                               Tracer tracer,
                                Transaction transaction,
                                ServerWebExchange exchange,
                                String description) {
 
         this.subscriber = subscriber;
         this.exchange = exchange;
         this.description = description;
-        this.tracer = tracer;
 
         transactionMap.put(this, transaction);
+
+        // store transaction into subscriber context it can be looked-up by reactor when the transaction
+        // is not already active in current thread.
+        this.context = subscriber.currentContext().put(AbstractSpan.class, transaction);
+    }
+
+    @Override
+    public Context currentContext() {
+        return context;
+    }
+
+    @Override
+    public void request(long n) {
+        subscription.request(n);
+    }
+
+    @Override
+    public void cancel() {
+        subscription.cancel();
+        cancelTransaction();
```

Comment:
[for reviewer] When `onSubscribe` is called, we capture the original subscription and delegate in order to handle cancellation directly without extra callback.